### PR TITLE
Easy forEach operation

### PR DESCRIPTION
**Pull request by sekaiamber**
## TL;DR;

Make an easy `forEach` operation by using a simple regular expression.
## sekaiamber

[[@sekaiamber](https://github.com/sekaiamber)]
## Extra

Close the PR #124 , using the new PR rules.

### CODE REVIEW

**Comment by sekaiamber** [comment]
Related #124, using the new PR rules.

**Comment by zenopopovici** [comment]
Thank you @sekaiamber, we'll put it under review next week when the new rules go into effect.

File: _posts/en/2016-xx-xx-easy-foreach-operation.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+tip-username: sekaiamber
+tip-username-profile: https://github.com/sekaiamber
+tip-tldr: Make an easy `forEach` operation by using a simple regular expression.
+
+categories:
+    - en
+---
+
+We can use a simple regular expression `/[^, ]+/g` to make an easy `forEach` operation.  
+For example, in jQuery, we can find a lot of DOM manipulations in pairs, like `append` and `appendTo`, they are just exchange the matched elements and target elements. So, we can code like this:
```
Comment (zenopopovici) [inline _posts/en/2016-xx-xx-easy-foreach-operation.md:15]: I'm not a fan of using jQuery for tips. If there is a specific jQuery issue, solved via a tip, I'm fine with that (e.g there is an angularJS related tip).

File: _posts/en/2016-xx-xx-easy-foreach-operation.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+"append,prepend".replace(/[^, ]+/g, function (name) {
+    jQuery[name + "To"] = function (selector) {
+        jQuery(selector)[name](this);
+        // ...
+    }
+});
+```
+
+**Note:** jQuery use its own `each` function to do this, see [source](https://github.com/jquery/jquery/blob/master/src/manipulation.js#L450).
```
Comment (zenopopovici) [inline _posts/en/2016-xx-xx-easy-foreach-operation.md:36]: There are some spelling mistakes, please check. (e.g. "jQuery uses")

File: _posts/en/2016-xx-xx-easy-foreach-operation.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    jQuery[name + "To"] = function (selector) {
+        jQuery(selector)[name](this);
+        // ...
+    }
+});
+```
+
+**Note:** jQuery use its own `each` function to do this, see [source](https://github.com/jquery/jquery/blob/master/src/manipulation.js#L450).  
+
+Another example, usually we don't want to expose all function of an object to users, we can simply do like this:
```
Comment (zenopopovici) [inline _posts/en/2016-xx-xx-easy-foreach-operation.md:38]: There are some spelling mistakes, please check. Phrase doesn't make sense.

**Comment by zenopopovici** [comment]
Not sure this tip will pass review. It needs to be a JS tip and be grammatically correct.

**Comment by zenopopovici** [comment]
@sekaiamber We discussed this, unless it points out a jQuery specific issue, we can't accept jQuery, either convert to vanilla JS or close PR.

**Comment by kurtextrem** [comment]
@zenopopovici You've misunderstood. jQuery is used as example and is pretty common. Their technique is clever and in this case shows the point.
The "real" code in this tip is vanilla JS.

However, I think the title is not fitting.

**Comment by zenopopovici** [comment]
I get what the tip does, but it revolves around how jQuery does stuff. Actual code should be given first, techniques described, and then maybe show alternatives. Not the other way around. It's too crude and a bit hard to understand.

**Comment by kurtextrem** [comment]
Actually I think the order is fine, it just needs other text. Such as:
"When taking a deep look into jQuery, we find a clever trick to define multiple functions with a lot less manual writing. For example you won't find `appendTo` and `prependTo`'s method body. Why is that?
...jQuery code...

We can easily use this technique like so:
...the code...
"

Also, jQuery is only named "jQuery" (so it could be any other library; We already have a tip similar to the one here, see: https://github.com/loverajoel/jstips/blob/gh-pages/_posts/en/2016-01-15-even-simpler-way-of-using-indexof-as-a-contains-clause.md), and apart from `jQuery(selector)` the code snippet **is** vanilla JS. But that is irrelevant in this case, as it is used as "showcase code", not as "use this code".

**Comment by zenopopovici** [comment]
@loverajoel Maybe you can have a look, i might be wrong... but i'm still not convinced it's a good one.

**Comment by gromgit** [comment]
I have two issues with this tip:
1. The title is very confusing. How could applying a function over an array be any easier than `forEach`? It took a while to understand that a more accurate title would be **Easy `forEach` on delimited string lists**.
2. The content documents a slightly shorter, but less clear and more fragile alternative to:

``` js
"append, replace".split(/[, ]+/).forEach(function(name) {
    // ...
}
```

For the tip to work, you need to remember that the regex must match _the individual elements you're interested in_, so you'd usually be specifying a _negative match_ on the delimiters. You'd therefore have to start with a delimiter match expression, which is precisely what `split` consumes.

**Comment by loverajoel** [comment]
ping

**Comment by zenopopovici** [comment]
Closed due to inactivity.

**Pull request closed without merging**